sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpSender.cs

```diff
@@ -461,7 +461,7 @@ public override async Task CloseAsync(CancellationToken cancellationToken)
                 request.AmqpMessage.ApplicationProperties.Map[ManagementConstants.Request.AssociatedLinkName] = sendLink.Name;
             }
 
-            request.Map[ManagementConstants.Properties.SequenceNumbers] = new[] { sequenceNumber };
+            request.Map[ManagementConstants.Properties.SequenceNumbers] = new List<long>(1) { sequenceNumber };
```

Comment:
We need to revert this as well.

Reply:
@ShivangiReja thanks for taking the time to educate me! Highly appreciated. How does the team make sure that this is not accidentally changed in the future? Passing an array seems to be a critical part for this library to operate and should almost be considered some type of public contract that cannot be broken. So I would have hoped a test would fail if I invalidate such important assumptions. Or at least have some kind of an approval test that verifies the output map would have been great I think.

Reply:
We do have live tests for Service Bus SDK which cover almost all the scenarios and this [test](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/tests/Sender/SenderLiveTests.cs#L170) would fail if you try to pass list instead of array.

If you would like to setup and run the live tests, you can follow this [guide](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/CONTRIBUTING.md#development-environment-setup).

Reply:
Forgive my ignorance but why was the PR green then (before my force push)? Are those tests not run on your CI/CD environment?

Reply:
Yeah, currently live tests don't run as a part of CI, only unit tests run because there isn't a recording test framework for AMQP as of now.

You can setup and run live tests locally by following this [guide](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/CONTRIBUTING.md#development-environment-setup).